internal/pkg/instrumentation/probe/interfaces.go

```diff
@@ -0,0 +1,73 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package probe provides instrumentation probe types and definitions.
+package probe
+
+import (
+	"log/slog"
+)
+
+// BaseProbe is the most basic type of Probe. It must support configuration,
+// loading and attaching eBPF programs, running, and closing.
+// All Probes must implement this Base functionality.
+type BaseProbe interface {
+	// ID is a unique identifier for the Probe.
+	// All types of Probes must provide a unique identifier for the framework.
+	ID() ID
+
+	// GetLogger returns an *slog.Logger for this Probe.
+	GetLogger() *slog.Logger
+
+	// Load loads the eBPF programs and maps required by the Probe into memory.
+	// The specific types of programs and maps are implemented by the Probe.
+	Load() error
```

Comment:
How do you specify what to load?

How will this handle multiple load targets when we support https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/197

Reply:
I wanted to see if I could get rid of all the arguments to `Load()` and `Attach()` to make them as generic as possible. To do that, I moved loading options into the Probe's object. But, I see now that doing so actually binds us to 1 process per 1 probe, which isn't what we want.